lib/line-index/src/lib.rs

```diff
@@ -177,10 +177,18 @@ impl LineIndex {
         Some(LineCol { line: line_col.line, col })
     }
 
+    /// Returns the given line's range.
+    pub fn line(&self, line: u32) -> Option<TextRange> {
+        let start = self.start_offset(line as usize)?;
+        let next_newline = self.newlines.get(line as usize).copied().unwrap_or(self.len);
+        let line_length = next_newline - start;
+        Some(TextRange::new(start, start + line_length))
+    }
+
     /// Given a range [start, end), returns a sorted iterator of non-empty ranges [start, x1), [x1,
     /// x2), ..., [xn, end) where all the xi, which are positions of newlines, are inside the range
     /// [start, end).
-    pub fn lines(&self, range: TextRange) -> impl Iterator<Item = TextRange> + '_ {
+    pub fn lines_in_range(&self, range: TextRange) -> impl Iterator<Item = TextRange> + '_ {
```

Comment:
This is a semver breaking change to the signatures so we'd need to release a `0.2.0` release then if we rename this function, should be fine to keep it as `lines`